code/game/mecha/equipment/weapons/weapons.dm

```diff
@@ -12,15 +13,30 @@
 			return 1
 	return 0
 
+/obj/item/mecha_parts/mecha_equipment/weapon/become_damaged()
+	if(!damaged)
+		damaged = 1
+		desc += "/nIt doesn't look to be in the best shape."
```

Comment:
I noticed there's still a lot of copypaste for this.

You should put lines 17, 18 and 19 on /atom/movable/proc/become_damaged() and call parent on the remaining become_damaged() that do something else afterwards.